hoodie-utilities/src/main/java/com/uber/hoodie/utilities/transform/Transformer.java

```diff
@@ -0,0 +1,25 @@
+package com.uber.hoodie.utilities.transform;
+
+import com.uber.hoodie.common.util.TypedProperties;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+
+/**
+ * UDF to transform source to target dataset before writing
+ */
+public interface Transformer {
+
+  /**
+   * Transform source RDD to target RDD
+   * @param jsc             JavaSparkContext
+   * @param sourceRdd       Source RDD
+   * @param sourceSchema    Source Schema
+   * @param targetSchema    Target Schema
+   * @param properties      Config properties
+   * @return Transformed RDD
+   */
+  JavaRDD<GenericRecord> apply(JavaSparkContext jsc, JavaRDD<GenericRecord> sourceRdd,
```

Comment:
I think it may be best to support DataFrame level transformations? Lets chat f2f

Reply:
@n3nash: Yes, this PR is intended to provide the transformation capability. I am going to make this PR as WIP in order to support DF.  Will ping you again once the PR is ready for review. 

Thanks,
Balaji.V